src/store/modules/lib/agents.js

```diff
@@ -57,11 +60,91 @@ export default () => ({
 		[UPDATE_USER_AGENT_TEXT] (state, { userAgentId, text }) {
 			state.userAgents.find(a => a.id === userAgentId).text = text
 		},
-		[CREATE_USER_AGENT] (state, { id, text, label }) {
-			state.userAgents.unshift({
-				id,
-				label,
-				text
+		[CREATE_USER_AGENT] (state, { id, text, label, isShared }) {
+			if (!isShared) {
+				state.userAgents.unshift({
+					id,
+					label,
+					text
+				})
+			} else {
+				let findOldAgentText = state.userAgents.find(a => a.text === text)
+				let findOldAgentLabel = state.userAgents.find(a => a.label === label)
+				if (findOldAgentText) {
+					findOldAgentText.text = text
+					findOldAgentText.id = id
+					// This below condition checks. If there is already named label and the old text and if that text === label's state text
+				} else if (findOldAgentLabel && findOldAgentText && findOldAgentLabel.text === findOldAgentText.text) {
+					findOldAgentLabel.text = text
+					findOldAgentLabel.id = id
+				} else if (findOldAgentLabel) {
+					var inBrackets = findOldAgentLabel.label.trim().match(/\(.*?\)/g)
+					if (findOldAgentLabel || inBrackets) {
+						if (inBrackets) {
+							var wasLabel = findOldAgentLabel.label
```

Comment:
consider using the existing function getIncrementedLabel instead

Reply:
The existing function cannot be used right now. But in future with time, I will try to manage it.